### PR TITLE
[Java Integration]: Fix sample

### PR DESCRIPTION
**Pull request by arthurscchan**
Fix sample testcases and readme to demonstrate how to reproduce the call graph generation process and static analysis for java program.

### CODE REVIEW

File: frontends/java/README.md [hunk 1]
```diff
@@ -7,6 +7,32 @@ Download and install java / maven in ubuntu
 sudo apt-get install -y openjdk-8-jdk-headless maven
```
Comment (AdamKorcz) [inline frontends/java/README.md:7]: Please format as code.

File: frontends/java/README.md [hunk 2]
```diff
@@ -17,7 +43,11 @@ It requires the target source code compiled and packed into jar file.
 
 The resulting call tree are shown in stdout.
 
-Example of running: `java -jar javacg-0.1-SNAPSHOT-static.jar <TARGET_JAR_FILE>`
+Command: `java -jar javacg-0.1-SNAPSHOT-static.jar <TARGET_JAR_FILE>`
+
+Example for execution using testcase test1: `java -jar javacg-0.1-SNAPSHOT-static.jar path/to/fuzz-introspector/tests/java/test1/test1.jar`
```
Comment (AdamKorcz) [inline frontends/java/README.md:48]: I think we need complete instructions here including getting `javacg-0.1-SNAPSHOT-static.jar`

File: frontends/java/README.md [hunk 3]
```diff
@@ -30,7 +60,13 @@ Depends on IBM's WALA https://github.com/wala/WALA, the maven build process will
 
 The resulting call tree are shown in stdout.
 
-Example of running: `./run.sh <-j | --jarFile> <jarFile1:...:javaFileN> <-e | --entryclass> <Public Entry Class Name>`
+**Current limitation, the entryclass must contains the main method to build the callgraph.**
+
+Example of running: `./run.sh <-j | --jarfile> <jarFile1:...:javaFileN> <-c | --entryclass> <Public Entry Class Name>`
```
Comment (AdamKorcz) [inline frontends/java/README.md:65]: Please include `cd frontends/java/wala`

File: frontends/java/README.md [hunk 4]
```diff
@@ -43,5 +79,158 @@ Depends on IBM's WALA https://github.com/soot-oss/soot, the maven build process
 
 The resulting call tree are shown in stdout.
 
-Example of running: `./run.sh <-j | --jarFile> <jarFile1:...:javaFileN> <-c | --entryclass> <Public Entry Class Name> <-m | --entrymethod <Public Entry Method Name>`
+Example of running: `./run.sh <-j | --jarfile> <jarFile1:...:javaFileN> <-c | --entryclass> <Public Entry Class Name> <-m | --entrymethod <Public Entry Method Name>`
```
Comment (AdamKorcz) [inline frontends/java/README.md:82]: Please include `cd frontends/java/soot`

File: frontends/java/README.md [hunk 5]
```diff
@@ -30,7 +60,13 @@ Depends on IBM's WALA https://github.com/wala/WALA, the maven build process will
 
 The resulting call tree are shown in stdout.
 
-Example of running: `./run.sh <-j | --jarFile> <jarFile1:...:javaFileN> <-e | --entryclass> <Public Entry Class Name>`
+**Current limitation, the entryclass must contains the main method to build the callgraph.**
+
+Example of running: `./run.sh <-j | --jarfile> <jarFile1:...:javaFileN> <-c | --entryclass> <Public Entry Class Name>`
+
+Example for execution using testcase test1: `./run.sh --jarfile path/to/fuzz-introspector/tests/java/test1/test1.jar --entryclass TestFuzzer`
```
Comment (AdamKorcz) [inline frontends/java/README.md:67]: For test1 I get:

When I run
```console
./run.sh --jarfile ../../../tests/java/test1/test1.jar --entryclass TestFuzzer
```

... I get
```console
Exception in thread "main" java.lang.IllegalStateException: java.nio.file.NoSuchFileException: /usr/lib/jvm/java-11-openjdk-amd64/jmods
        at com.ibm.wala.util.PlatformUtil.getBootClassPathJars(PlatformUtil.java:64)
        at com.ibm.wala.properties.WalaProperties.getJ2SEJarFiles(WalaProperties.java:69)
        at com.ibm.wala.core.util.config.AnalysisScopeReader.processScopeDefLine(AnalysisScopeReader.java:221)
        at com.ibm.wala.core.util.config.AnalysisScopeReader.read(AnalysisScopeReader.java:103)
        at com.ibm.wala.core.util.config.AnalysisScopeReader.readJavaScope(AnalysisScopeReader.java:70)
        at com.ibm.wala.core.util.config.AnalysisScopeReader.makePrimordialScope(AnalysisScopeReader.java:245)
        at com.ibm.wala.core.util.config.AnalysisScopeReader.makeJavaBinaryAnalysisScope(AnalysisScopeReader.java:258)
        at ossf.fuzz.introspector.wala.JarCallGraphGenerator.main(JarCallGraphGenerator.java:59)
Caused by: java.nio.file.NoSuchFileException: /usr/lib/jvm/java-11-openjdk-amd64/jmods
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(UnixFileSystemProvider.java:432)
        at java.base/java.nio.file.Files.newDirectoryStream(Files.java:472)
        at java.base/java.nio.file.Files.list(Files.java:3699)
        at com.ibm.wala.util.PlatformUtil.getBootClassPathJars(PlatformUtil.java:59)
        ... 7 more
```

File: frontends/java/README.md [hunk 6]
```diff
@@ -43,5 +79,158 @@ Depends on IBM's WALA https://github.com/soot-oss/soot, the maven build process
 
 The resulting call tree are shown in stdout.
 
-Example of running: `./run.sh <-j | --jarFile> <jarFile1:...:javaFileN> <-c | --entryclass> <Public Entry Class Name> <-m | --entrymethod <Public Entry Method Name>`
+Example of running: `./run.sh <-j | --jarfile> <jarFile1:...:javaFileN> <-c | --entryclass> <Public Entry Class Name> <-m | --entrymethod <Public Entry Method Name>`
+
+Example for execution using testcase test1: `./run.sh -j path/to/fuzz-introspector/tests/java/test1/test1.jar -c TestFuzzer -m fuzzerTestOneInput`
```
Comment (AdamKorcz) [inline frontends/java/README.md:84]: I get:

```console
Exception in thread "main" java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.Exception: Error: The path '/usr/lib/jvm/java-11-openjdk-amd64/lib/rt.jar' does not exist.
        at soot.SourceLocator.getClassSourceType(SourceLocator.java:312)
        at soot.SourceLocator.lookupInClassPath(SourceLocator.java:615)
        at soot.asm.AsmClassProvider.find(AsmClassProvider.java:40)
        at soot.SourceLocator.getClassSource(SourceLocator.java:187)
        at soot.SootResolver.bringToHierarchyUnchecked(SootResolver.java:250)
        at soot.SootResolver.bringToHierarchy(SootResolver.java:240)
        at soot.SootResolver.bringToSignatures(SootResolver.java:310)
        at soot.SootResolver.bringToBodies(SootResolver.java:350)
        at soot.SootResolver.processResolveWorklist(SootResolver.java:187)
        at soot.SootResolver.resolveClass(SootResolver.java:152)
        at soot.Scene.loadClass(Scene.java:1004)
        at ossf.fuzz.introspector.soot.CallGraphGenerator.main(CallGraphGenerator.java:72)
Caused by: java.util.concurrent.ExecutionException: java.lang.Exception: Error: The path '/usr/lib/jvm/java-11-openjdk-amd64/lib/rt.jar' does not exist.
        at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:552)
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:513)
        at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:90)
        at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:197)
        at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2229)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2195)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2153)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2043)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3851)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3875)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4800)
        at soot.SourceLocator.getClassSourceType(SourceLocator.java:310)
        ... 11 more
Caused by: java.lang.Exception: Error: The path '/usr/lib/jvm/java-11-openjdk-amd64/lib/rt.jar' does not exist.
        at soot.SourceLocator$1.load(SourceLocator.java:73)
        at soot.SourceLocator$1.load(SourceLocator.java:68)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3445)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2194)
        ... 17 more
```

File: frontends/java/README.md (hunk 1 above)
Comment (arthurscchan) [reply frontends/java/README.md]: Fixed.

File: frontends/java/README.md (hunk 2 above)
Comment (arthurscchan) [reply frontends/java/README.md]: Added with details description.

File: frontends/java/README.md (hunk 3 above)
Comment (arthurscchan) [reply frontends/java/README.md]: Fixed.

File: frontends/java/README.md (hunk 4 above)
Comment (arthurscchan) [reply frontends/java/README.md]: Added.

File: frontends/java/README.md (hunk 6 above)
Comment (arthurscchan) [reply frontends/java/README.md]: Fixed.

**Review by DavidKorczynski — APPROVED** [review]

**Comment by arthurscchan** [comment]
Related to 2 and 3 of issue #536

**Pull request merged**